data/trie/hashesHolder/checkpointHashesHolder.go

```diff
@@ -77,16 +80,19 @@ func (c *checkpointHashesHolder) RemoveCommitted(lastCommittedRootHash []byte) {
 	defer c.mutex.Unlock()
 
 	sizeOfRemovedHashes := uint64(0)
-	for index, hashesMap := range c.hashes {
-		sizeOfRemovedHashes = sizeOfRemovedHashes + getMapSize(hashesMap, c.hashSize)
+	for index, rootHash := range c.rootHashes {
+		mapHashes := c.hashes[index]
+		sizeOfRemovedHashes = sizeOfRemovedHashes + getMapSize(mapHashes, c.hashSize) + uint64(len(rootHash))
 
-		_, ok := hashesMap[string(lastCommittedRootHash)]
-		if !ok {
+		lastCommittedRootHashNotFound := !bytes.Equal(c.rootHashes[index], lastCommittedRootHash)
```

Comment:
Why not directly rootHash instead c.rootHashes[index]

Reply:
fixed